lib/Backends/OpenCL/OpenCL.cpp

```diff
@@ -692,8 +713,9 @@ void OCLBackend::doForwardPass() {
           createKernel(useTiledMatMul ? "matmul_tiled" : kernelName);
       setKernelArg(kernel, 0, deviceBuffer_);
 
-      unsigned numArgs = I.getNumOperands();
-      for (unsigned arg = 0; arg < numArgs; arg++) {
+      /* unsigned numArgs = I.getNumOperands(); */
```

Comment:
What is this?

Reply:
An bug, test case show it was 4, but we expected 3

Reply:
@opti-mix Do you mind looking at this bug?

Reply:
@lygstate I checked under debugger and I see I.getNumOperands() == 3 for MatMulInst instructions. It is also defined with 3 arguments in InstrGen.cpp. 

If you have discovered a case, where it reported 4 arguments, could you provide a description how to reproduce this?

Reply:
```

TEST_P(Operator, simplePredication) {
  auto *inputs = mod_.createVariable(ElemKind::FloatTy, {10, 10, 10}, "inputs");
  auto *counters = mod_.createVariable(ElemKind::FloatTy, {10}, "counters");

  counters->getHandle() = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
  inputs->getHandle().randomize(-10, 10, mod_.getPRNG());

  Node *C5 = F_->createSplat("C5", counters->getType(), 5.0);
  Node *pred = F_->createCmpLTE("cmp", C5, counters);

  auto *FC0 = F_->createFullyConnected("FC0", inputs, 128);
  auto *RL0 = F_->createRELU("RL0", FC0);
  auto *FC1 = F_->createFullyConnected("FC1", RL0, 64);
  auto *RL1 = F_->createRELU("RL1", FC1);
  auto *FC2 = F_->createFullyConnected("FC2", RL1, 32);
  auto *RL2 = F_->createRELU("RL2", FC2);

  F_->createSave("ret", RL2);

  FC0->setPredicate(pred);
  FC1->setPredicate(pred);
  FC2->setPredicate(pred);

  EE_.compile(CompilationMode::Infer, F_);
  EE_.run({}, {});
}
```
The OperatorTest.cpp
simplePredication

Reply:
I have already use the CI to trigger the issue